Sources/StreamChat/Repositories/SyncOperations.swift

```diff
@@ -104,23 +104,44 @@ final class RefetchChannelListQueryOperation: AsyncOperation {
                 watchedAndSynchedChannelIds: context.watchedAndSynchedChannelIds,
                 synchedChannelIds: context.synchedChannelIds
             ) { result in
-                switch result {
-                case let .success((watchedChannels, unwantedCids)):
-                    log.info("Successfully refetched query for \(query.debugDescription)", subsystems: .offlineSupport)
-                    let queryChannelIds = watchedChannels.map(\.cid)
-                    context.watchedAndSynchedChannelIds = context.watchedAndSynchedChannelIds.union(queryChannelIds)
-                    context.unwantedChannelIds = context.unwantedChannelIds.union(unwantedCids)
-                    done(.continue)
-                case let .failure(error):
-                    log.error(
-                        "Failed refetching query for \(query.debugDescription): \(error)",
-                        subsystems: .offlineSupport
-                    )
-                    done(.retry)
-                }
+                Self.handleReset(query, with: result, context: context, done: done)
             }
         }
     }
+    
+    init(queryProvider: @escaping () -> ChannelListQuery?, updater: ChannelListUpdater, context: SyncContext) {
```

Comment:
Since we have a new type then I had to make sure that if someone uses that, then the sync repo resets channels in that case as well. The new type, ChatList is a struct, so that is why I did this closure based approach. Also, ChatList is 13.0 or later, I can't use the type in this code without a lot of available checks.